#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.4055384939685732 0.4259000137266124 -0.8087939837915296 1.733291636007106
  position 0.02640710963681614 4.719639897028212 5.988594179208095
}
TexturedBackground {
}
RectangleArena {
  floorSize 5 5
}
Robot {
  translation -0.39 2.16 0.0400079
  rotation 1 0 0 1.5708
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 0.707108281185553 1.0171504044755327e-08 0.70710528118436 3.1415853071795863
        children [
          DEF WHEEL Pose {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.305882 0.898039 0.25098
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.04
                  subdivision 24
                }
              }
            ]
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 0 -0.9999999999999999 -2.355139888694314e-16 1.570795307179586
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 1.3345761387975192e-15 1 0 1.5707999999999984
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 1.5700932591295458e-16 -0.9999999999999999 3.9252331478238657e-16 1.5707953071795813
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    LightSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.5
            metalness 0.4
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_left"
    }
    LightSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.5
            metalness 0.4
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_right"
    }
  ]
  name "robot 3b"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "Braitenberg_Vehicle"
}
Robot {
  translation -0.83 1.44 0.040008
  rotation 1 0 0 1.5708
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 0.707108281185553 1.0171504044755327e-08 0.70710528118436 3.1415853071795863
        children [
          DEF WHEEL Pose {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.305882 0.898039 0.25098
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.04
                  subdivision 24
                }
              }
            ]
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 0 -0.9999999999999999 -2.355139888694314e-16 1.570795307179586
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 1.3345761387975192e-15 1 0 1.5707999999999984
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 1.5700932591295458e-16 -0.9999999999999999 3.9252331478238657e-16 1.5707953071795813
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    LightSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.5
            metalness 0.4
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_left"
    }
    LightSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.5
            metalness 0.4
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_right"
    }
  ]
  name "robot 2a"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "Braitenberg_Vehicle"
}
Robot {
  translation 1.58 2.16 0.04
  rotation 1 0 0 1.5708
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 0.707108281185553 1.0171504044755327e-08 0.70710528118436 3.1415853071795863
        children [
          DEF WHEEL Pose {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.305882 0.898039 0.25098
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.04
                  subdivision 24
                }
              }
            ]
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 0 -0.9999999999999999 -2.355139888694314e-16 1.570795307179586
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 1.3345761387975192e-15 1 0 1.5707999999999984
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 1.5700932591295458e-16 -0.9999999999999999 3.9252331478238657e-16 1.5707953071795813
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    LightSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.5
            metalness 0.4
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_left"
    }
    LightSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.5
            metalness 0.4
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_right"
    }
  ]
  name "robot 2b"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "Braitenberg_Vehicle"
}
Robot {
  translation 0.37 2.16 0.04
  rotation 1 0 0 1.5708
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 0.707108281185553 1.0171504044755327e-08 0.70710528118436 3.1415853071795863
        children [
          DEF WHEEL Pose {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.305882 0.898039 0.25098
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.04
                  subdivision 24
                }
              }
            ]
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 0 -0.9999999999999999 -2.355139888694314e-16 1.570795307179586
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 1.3345761387975192e-15 1 0 1.5707999999999984
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 1.5700932591295458e-16 -0.9999999999999999 3.9252331478238657e-16 1.5707953071795813
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    LightSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.5
            metalness 0.4
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_left"
    }
    LightSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.57
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 0.5
            metalness 0.4
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_right"
    }
  ]
  name "robot 3a"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "Braitenberg_Vehicle"
}
PointLight {
  attenuation 0 0 1
  location 0 0 0.53
}
